SpaceHorse/SpaceHorse/Space_display.cpp

```diff
@@ -119,120 +119,163 @@ void Space_display::update()
+	*/
 	
-	//Draw all 9 of the background sprites, tiled around, and including, the current grid
-	int backX = m_gridX-m_windowWidth; //X coordinate of top left corner tile of the 9 tiles
-	int backY = m_gridY-m_windowHeight; //Y coordinate of top left corner tile of the 9 tiles
-	for (int x=0; x<3; x++)
+	drawBackground();
+	drawPlanets();
+	drawSpaceJunk();
+	drawShips();
+	drawBorder();
+	drawStats();
+	drawRadar();
+
+	//Game over
+	if (m_gameOver)
+	{ 
+		al_draw_text(font, al_map_rgb(250, 0, 20), m_windowWidth*0.5, m_windowHeight*0.5, ALLEGRO_ALIGN_CENTRE, "GAME OVER"); 
+	}
+
+	//Hyper Drive input
+	if (m_hyperDrive)
 	{
-		for (int y=0; y<3; y++)
-		{ al_draw_bitmap(m_backgroundSprite, backX + m_windowWidth*x - m_bgX, backY + m_windowHeight*y - m_bgY,0); }
+		al_draw_text(font, al_map_rgb(255, 255, 255), m_windowWidth*0.5, m_windowHeight*0.35, ALLEGRO_ALIGN_CENTRE, "Enter coordinates: ");
+		al_draw_text(font, al_map_rgb(255, 255, 255), m_windowWidth*0.5, m_windowHeight*0.4, ALLEGRO_ALIGN_CENTRE, m_editText.c_str());
 	}
+}
 
-	//Draw planets
-	for (int i=0; i<(*m_map).getPlanets().size(); i++)
+void Space_display::drawBackground(void)
+{
+	//Draw all 9 of the background sprites, tiled around, and including, the current grid
+	int backX = m_gridX - m_windowWidth; //X coordinate of top left corner tile of the 9 tiles
+	int backY = m_gridY - m_windowHeight; //Y coordinate of top left corner tile of the 9 tiles
+	for (int x = 0; x<3; x++)
+	{
+		for (int y = 0; y<3; y++)
+		{
+			al_draw_bitmap(m_backgroundSprite, backX + m_windowWidth*x - m_bgX, backY + m_windowHeight*y - m_bgY, 0);
+		}
+	}
+}
+void Space_display::drawPlanets(void)
+{
+	for (int i = 0; i<(*m_map).getPlanets().size(); i++)
 	{
 		al_draw_tinted_scaled_rotated_bitmap(m_planetSprites.at(i),
-			al_map_rgb((*m_map).getPlanets().at(i).getColor1(), (*m_map).getPlanets().at(i).getColor2(), (*m_map).getPlanets().at(i).getColor3()), 
-			getPlanetSpriteWidth(i)/2, getPlanetSpriteHeight(i)/2,
-			(*m_map).getPlanets().at(i).getX()-m_bgX, (*m_map).getPlanets().at(i).getY()-m_bgY, 
+			al_map_rgb((*m_map).getPlanets().at(i).getColor1(), (*m_map).getPlanets().at(i).getColor2(), (*m_map).getPlanets().at(i).getColor3()),
+			getPlanetSpriteWidth(i) / 2, getPlanetSpriteHeight(i) / 2,
+			(*m_map).getPlanets().at(i).getX() - m_bgX, (*m_map).getPlanets().at(i).getY() - m_bgY,
 			(*m_map).getPlanets().at(i).getPlanetScale(), (*m_map).getPlanets().at(i).getPlanetScale(), 0, 0);
 	} //Draw planets only if their coordinates exist within current screen
-	
-	//Draw sprites for each Ship object
-	for (int i=0; i<(*m_players).size(); ++i)
+}
+void Space_display::drawSpaceJunk(void)
+{
+	// TODO: Only draw space junk if on screen
+	for (int i = 0; i < m_map->getSpaceJunk().size(); i++)
+	{
+		ALLEGRO_BITMAP *sprite = m_map->getSpaceJunk().at(i).getSprite();
+		al_draw_tinted_scaled_rotated_bitmap(sprite,
+			al_map_rgb(255, 255, 255),
+			al_get_bitmap_width(sprite), al_get_bitmap_height(sprite),
```

Comment:
These two parameters, sprite *width* and *height*, are supposed to be used to specify the centre of the sprite (check the allegro method API :+1: ), so the sprite can be drawn from the centre - useful when using the *al_draw_tinted_scaled_***rotated***_bitmap* method (obviously, you need a centre to rotate an image).
You should be dividing the width/height returned in half when passing in as a parameter here. Otherwise, the image won't be drawn where you expect it - this is happening here, and the junk gets collected at a weird offset to the bottom right.